### PR TITLE
PR 2014 - feature/AT-9828-has-funding-page

### PR DESCRIPTION
**Pull request by AndrewNickerl-ccpo**
Complete CurrentlyHasFunding page with updates to routing and store
Screenshot 2023-10-04 at 1 06 03 PM

### CODE REVIEW

File: src/steps/10-FinancialDetails/CurrentlyHasFunding.spec.ts [hunk 1]
```diff
@@ -28,4 +34,58 @@ describe("Testing CurrentlyHasFunding component", () => {
...
+      (AcquisitionPackage as any).isDitcoUser = jest.fn().mockReturnValue(false);
+
+      const pushMock = jest.fn();
+      wrapper.vm.$router = { push: pushMock } as any;
+
+      wrapper.vm.addNavigation();
+
+      expect(pushMock).toHaveBeenCalledWith({
+        name: routeNames.RFD,
+      });
```
Comment (tonysumner-ccpo) [inline src/steps/10-FinancialDetails/CurrentlyHasFunding.spec.ts:88]: try not to use `any`
This code should work....
```
 it("should navigate to RFD page if non-DITCO user", () => {
      jest.spyOn(wrapper.vm, "isDitcoUser").mockReturnValue(false);

      const pushMock = jest.fn();
      wrapper.vm.$router = { push: pushMock as RawLocation } as VueRouter;
      wrapper.vm.addNavigation();

      expect(pushMock).toHaveBeenCalledWith({
        name: routeNames.RFD,
      });
    });
```

File: src/router/resolvers/index.ts [hunk 2]
```diff
@@ -1552,11 +1552,18 @@ export const IGCESupportingDocumentationResolver = (current: string): string =>
...
+    : isDitcoUser()
+      ? routeNames.CurrentlyHasFunding
+      : routeNames.RFD
 };
 
+export const GTCInformationResolver = (current: string): string => {
+  return FinancialDetails.hasFunding === "HAS_FUNDING"
+    ? routeNames.GTC
+    : routeNames.GeneratingPackageDocuments
+}
```
Comment (tonysumner-ccpo) [inline src/router/resolvers/index.ts:1566]: use `routeNames.routeNames.GeneratingPackageDocumentsFunding` instead.  `routeNames.GeneratingPackageDocuments` navigates the user to the components in step 9 and then if the user clicks back/continue they remain in step 9

File: src/steps/10-FinancialDetails/CurrentlyHasFunding.vue [hunk 3]
```diff
@@ -1,30 +1,107 @@
...
+      label: "Yes. I have a GT&C, as well as a FS Form 7600B or MIPR.",
+      value: "HAS_FUNDING"
+    },
+    {
+      id: "NoFundingCheckbox",
+      label:
+          "No. I need to obtain funding documents.<br/>" +
+          "<span style='color: #41494E'>We’ll generate a draft of your package to " +
+          "submit to your financial office next.</span>",
+      value: "NO_FUNDING",
```
Comment (tonysumner-ccpo) [inline src/steps/10-FinancialDetails/CurrentlyHasFunding.vue:62]: remove the span in line 60, so the error class will decorate all text...
image

File: src/steps/10-FinancialDetails/CurrentlyHasFunding.spec.ts (hunk 1 above)
Comment (AndrewNickerl-ccpo) [reply src/steps/10-FinancialDetails/CurrentlyHasFunding.spec.ts:88]: Removed the `any` from the router but had to keep it on the `isDitcoUser` mock due to the way we export the function

File: src/steps/10-FinancialDetails/CurrentlyHasFunding.vue (hunk 3 above)
Comment (AndrewNickerl-ccpo) [reply src/steps/10-FinancialDetails/CurrentlyHasFunding.vue:61]: Removed the span and verified that all text now turns red upon validation failure

File: src/router/resolvers/index.ts (hunk 2 above)
Comment (AndrewNickerl-ccpo) [reply src/router/resolvers/index.ts:1566]: Updated resolver to use `routeNames.GeneratingPackageDocumentsFunding`

**Comment by tonysumner-ccpo** [comment]
in order for the unit tests to pass, an adjustment had to be made to `src\store\attachments\index.ts`
Please upload at least 1 document to each of the following pages, then exit and reenter the package then revisit each of these pages to ensure the documents persists.

- [ ] /current-contract/upload-charts
- [ ] /current-contract/upload-process
- [ ] /requirements-cost-estimate/supporting-documentation
- [ ] /requirements-cost-estimate/mipr
- [ ] /requirements-cost-estimate/upload-7600
- [ ] /upload-ja-mrr-documents/upload-signed-documents

**Review by DGabriel-CCPO — APPROVED** [review]

**Pull request merged**